### ISSUE TITLE
How to generate images from the given description?

### ISSUE THREAD

**Issue by ozamanan** [weihaox/TediGAN#5]
Hello, I had a doubt. How do we generate images directly from text descriptions. I executed the invert_v2.py code and it seems that it manipulates an input image.

**Comment by weihaox** [comment]
The given image is first mapped into the latent space of a pretrained StyleGAN model to obtain its latent code. If the latent code is not obtained from a given image and is sampled randomly (from a normal distribution), you can generate images directly from text descriptions. To be specific,  this [line](https://github.com/weihaox/TediGAN/blob/main/idinvert_pytorch/utils/inverter.py#L184): `init_z = self.get_init_code(image)` should be replaced with the randomly sampled vectors of the same size. 

I have been occupied these days and will add the corresponding codes soon. I would appreciate it if you could help with this.

**Comment by ozamanan** [comment]
I followed the step you mentioned but I am getting the following image as a result. I ran the model for 1000 iterations and the description was "This man is smiling. He is young." with the same learning rate
![142_inv](https://user-images.githubusercontent.com/21983436/110139446-06124780-7da1-11eb-88f9-3c2012b9d6c4.png)

**Comment by weihaox** [comment]
Hi @ozamanan Thank you for sharing the result. 

I ran the model with the description "This man is smiling. He is young" and used the default setting (200 iterations). The result is shown below. 
![image](https://user-images.githubusercontent.com/54136324/110190395-11777880-7e5e-11eb-90dd-49bdc5f1dc9e.png)
The result after 1000 iterations is 
![image](https://user-images.githubusercontent.com/54136324/110190568-e93c4980-7e5e-11eb-8b86-afc0aa195845.png)
The above results are quite creepy. I felt that the problem may be caused by the CLIP loss. So I changed the loss_weight_clip from 2.0 to 1.0 and got the following result.
![image](https://user-images.githubusercontent.com/54136324/110190677-6ebff980-7e5f-11eb-9c70-e0aa8b3c5e3d.png)
I didn't carefully choose the value of loss_weight_clip before. It seems that setting loss_weight_clip to 2.0 is a bit large. Changing loss_weight_clip may be more helpful to generate meaningful results, rather than increasing the number of iterations.

Besides, the CLIP hasn't released the training data. We do not know if it contains all the desired face attributes. The results are not always satisfactory when generating or editing images based on certain descriptions.

**Comment by ozamanan** [comment]
Yes, I got these results as well, they use the given example image and manipulate it. But I was talking about the generation of images directly from text descriptions, for which I got the above result. Did I miss anything for that?

**Comment by weihaox** [comment]
Sorry for my misunderstanding...Could you share these codes?

**Comment by ozamanan** [comment]
Thank you very much. This worked perfectly. Now I was just wondering how would you generate the 1024x1024 resolution images. The readme file has a pretrained model stylegan_gghq_1024.pth but does have a styleganinv_ffhq1024_encoder.pth and styleganinv_ffhq1024_generator.pth

**Comment by weihaox** [comment]
The basic idea is to invert an given image into the latent space of StyleGAN model (there are three directions for this, you can refer to our GAN inversion [survry](https://arxiv.org/abs/2101.05278) and a curated [list](https://github.com/weihaox/awesome-image-translation/blob/master/awesome-gan-inversion.md) of inversion papers). 

For 1024x1024 resolution, you can try other model pretrained on 1024 FFHQ like StyleGAN2Ada. Their official repo provides a projector.py that can directly obtain the inverted latent codes of an image. In this case, you don't need an encoder. 

Since the problems of _How to generate images from the given description_ have been solved, please close the issue . You can open another issue for deteiled discuss on 1024 image generation if you want.
